Dip/Dip/Definition.swift

```diff
@@ -50,20 +50,62 @@ func ==(lhs: DefinitionKey, rhs: DefinitionKey) -> Bool {
 public enum ComponentScope {
   /// Indicates that a new instance of the component will be created each time it's resolved.
   case Prototype
+  /// Indicates that instances will be reused during resolve but will be discurded when topmost `resolve` method returns.
+  case ObjectGraph
   /// Indicates that resolved component should be retained by container and always reused.
   case Singleton
 }
 
 ///Definition of type T describes how instances of this type should be created when they are resolved by container.
 public final class DefinitionOf<T>: Definition {
```

Comment:
Can't we do that?
```
container.register(.ObjectGraph) { Server() as Server }
  .resolveDependencies(container) { container, server in
    server.client = container.resolve() as Client
  }
```
Sure it might seem odd to chain a call after a trailing closure, but that works, and some might be ok with that style. I myself as a user of the lib I would prefer writing it like you did with two lines but others can choose to chain too, so I'm ok with that ;)

One thing newcomers might be wondering is why you call `resolveDependencies` on `serverDefinition` and not the `clientDefinition` of your ObjectGraph or both. Be sure to explain the rationale and logic of that in the playground.